### ISSUE TITLE
Feature request: optionally require explicit "no action this turn" for idle units

### ISSUE THREAD

**Issue by ghost** [yairm210/Unciv#6884]
Button "Next unit" goes through units only once. Especially later in the game, when you have tens of units, you may want to move them in certain order, or at least some units after the other. For example, if you want to attack in specific order. However, with the current UI that's not easy.

I can think of two alternatives:

1) Add a special no-action command to units, something like "Wait" (key `W` on desktop). When this pseudocommand is issued, unit gets skipped for now, but moved to the end of the queue of "Next unit". So, user intention expressed here would be sth. like "Yes, I want to move this unit, but not now, remind me again later, when I'm done with other units".

2) Make this explicit: require that all units are moved or explicit command like e.g. "Skip this turn", "Stay put this turn" etc. is always issued. In other words, until all units receive some command (or are fortified/sleeping, those would be ignored as now), button "Next unit" would not become "Next turn". This could be an optional mode activated on "Options" screen, so that by default it would work as now.

**Comment by SomeTroglodyte** [comment]
Not possible with the current data model. There's no 'queue', just a 'due' flag. And honestly, I can relate - I remember CTP's way - but this is minor. After all, having a brain, and using it to temporarily store a few "to-do"'s is a viable workaround? :grin: 

Your "2" might correspond to us not clearing 'due' as it is now (when you hit next when next just has selected your general and maybe a few other situations) but on-command only. Much easier to code, but - I don't think that would really help much. I may  be wrong..... Yes, maybe, I'd have to see it, my imagination isn't working. Volunteers?

**Comment by anton-koerbchen** [comment]
> Not possible with the current data model.

Well, it is there to be improved. Doesn't strike me as something terribly difficult to write.

> After all, having a brain, and using it to temporarily store a few "to-do"'s is a viable workaround? grin

It is, but a nicer UI that tries to prevent simple blunders is always better than saying: "Haha, moron, it's your fault!".

> I don't think that would really help much.

This would make the game enforce that you must make a decision for every unit before being able to finish the turn. I.e. either a unit must be fortified/sleeping, you need to move/attack etc. with it, or explicitly say "I don't want to use its [remaining] AP this turn". As I proposed above, this should be only an optional mode, because I'd presume not everyone would like it.

**Comment by SomeTroglodyte** [comment]
> moron, it's your fault

Now I'll _have_ to find a place in the game to output that quote... Maybe as easter egg... :rabbit:  :egg: 

> something terribly difficult to write

Go ahead, we'll help you along. And if you indeed start, _do_ tell us where (and how) our getting started documentation is lacking :grinning:

**Comment by SomeTroglodyte** [comment]
> > "Go ahead" 

> #6896 little later

:+1:

**Comment by SomeTroglodyte** [comment]
Solved, by @doublep
